### DISCUSSION TITLE
How to create manual mappers/middleware for Raw Queries in Prisma?

### DISCUSSION THREAD

**Discussion by Enforc3rr** [prisma/prisma#23427] [Q&A] [answered]
### Question

How can I create manual mappers for raw queries in prisma ?
The problem is, date time field in my database are stored in dd-MM-yyyy hh:mm:ss format ( by default they are in IST ( its a old DB and changing it isn't possible now )  ) , now when using the raw queries they are automatically being treated as UTC , which I want to prevent , I want to know how can I add a manual mapper that can fetch me the date data in string format which using any library I convert it to required date formats further. 

tldr ; I want a mapper/Middleware for a specific raw query, which can get me values in raw format (string) instead of converting its type.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS:  macOS
- Database:MySQL
- Node.js version: 20.11.0

- Run `prisma -v` to see your Prisma version and paste it
Environment variables loaded from .env
prisma                  : 5.10.2
@prisma/client          : 5.10.2
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.11.0
Query Engine (Node-API) : libquery-engine 5a9203d0590c951969e85a7d07215503f4672eb9 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 5a9203d0590c951969e85a7d07215503f4672eb9 (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.10.0-34.5a9203d0590c951969e85a7d07215503f4672eb9
Default Engines Hash    : 5a9203d0590c951969e85a7d07215503f4672eb9
Studio                  : 0.499.0

**Comment by ludralph** [comment, marked as answer]
Hi @Enforc3rr 👋 

Thank you for raisin this queston.

Prisma does not currently support custom mappers for raw queries. Prisma maps any database values returned by `$queryRaw` and `$queryRawUnsafe` to their corresponding JavaScript types. Prisma Client returns all `DateTime` as native [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) objects. This implies that it's not possible to return the Date in the string format you want as by default Prisma would return it as Date in UTC format.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Enforc3rr** [reply]
Thank you for the answer , if we can get a mapper like Java Lib JDBI , that would be great. but thanks.
